### PR TITLE
Implemented Tvshowid_int page

### PR DESCRIPTION
**Pull request by developingright**
Hi @razzeee,
I have created and implemented the `tvshowid_int` page which contains the information about a particular `tvShow`. I have worked on its routing, created a `SeasonObj`, modified decoders, created new decoders from scratch, and implemented the `viewSeason` card for the `SeasonObj`. Please let me know if any changes are required. 

| Chorus 2| Elm Chorus |
| --- | --- |
| Screenshot from 2022-08-12 20-02-26 | Screenshot from 2022-08-12 20-02-16 |

### CODE REVIEW

**Comment by razzeee** [comment]
Hrm, I can't get through the tvshow/movie list pages. It shows no movies/tvshows for me. 

The console log shows that it's receiving data. But `movie_list` / `tvshow_list` is empty, not sure if my library is just too big or if my data is different, as I might be using a different scrapper?

**Comment by developingright** [comment]
Hi @razzeee, 
I think if it’s working fine with Chorus 2 then it should also work with Elm Chorus. I am using `TMDb Tv Shows` and `TMDb Movie Scrapper`. I wonder if you are able to get a list of all tvShows/Movies in return if you make a JSON RPC request from `Kodi/labs/API-browser` in Chorus 2.

**Comment by razzeee** [comment]
I'm not near a pc to test that in the next 5 days

**Comment by razzeee** [comment]
http://localhost:8080/#lab/api-browser/VideoLibrary.GetMovies for e.g. returns fine on chorus 2 and it seems to work in the frontend too. while it breaks for elm-chorus.

I haven't replicated that for tvshows, but will do soon.

**Comment by razzeee** [comment]
The poster should be clickable in the overview too

**Comment by razzeee** [comment]
image

- Image on the right doesn't fade
- Long texts do not get truncated. The old chorus had a section that could be extended.

**Comment by developingright** [comment]
@razzeee, It should be clickable, like here in this screenshot, I wonder what could be causing it to not work. Also, the image on the right fades in Chrome and other supported browsers except for Mozilla, etc. the mask-image property isn't supported in Mozilla Firefox. I will make the other changes asap.

Screenshot from 2022-08-17 15-24-29
> The poster should be clickable in the overview too

**Comment by razzeee** [comment]
We need something that works in firefox, as that should be the browser we target. As it's the only OSS engine.

**Comment by razzeee** [comment]
To be clear this is the view, where I can't click the poster images. I need to click on the show title to get to the detail page.

image

**Comment by developingright** [comment]
Hi @razzeee,
 Since this change is related to card UI, I was thinking that creating a separate PR to make posters clickable and resolving the [issue](https://github.com/xbmc/elm-chorus/issues/114) "truncate the items" for the cards will be better.

In addition, regarding the image fade effect not working in Firefox, I was thinking of fixing UI irregularities and CSS bugs in the end once pages and functionalities have been implemented. 

It would be great if you could share your thoughts on this.

**Comment by razzeee** [comment]
First one makes sense.

I'm not so sure about the second one, but if you would like that that's fine. But will make it harder for me to test.

**Comment by developingright** [comment]
It should not make it hard for you to test as I will make sure to mention the changes I have made and the bugs / irregularities that I have fixed. 🙂 


> First one makes sense.
> 
> I'm not so sure about the second one, but if you would like that that's fine. But will make it harder for me to test.

**Comment by developingright** [comment]
Hi @razzeee, 
I have made the required change.

Screenshot from 2022-08-17 21-06-01
on click ->
Screenshot from 2022-08-17 21-06-09

File: public/main.css [hunk 1]
```diff
@@ -356,6 +357,22 @@ h2 {
...
 }
 
+#description {
+  display: none;
+}
+
+#description:checked + label {
+  height: 100%;
+}
+.description-text {
```
Comment (razzeee) [inline public/main.css:367]: this should probably have a hover cursor change, to hint at the fact that it's interactable

File: public/main.css [hunk 2]
```diff
@@ -333,6 +333,7 @@ h2 {
     rgba(0, 0, 0, 1),
```
Comment (razzeee) [inline public/main.css:333]: in the line above, for our case to work we need `to left`. Then firefox will be fine on that page. As it seems that firefox is the only browser correctly using the unprefixed css and chromiums are stuck with the prefixed ones.

Comment (developingright) [reply public/main.css:333]: Done :+1: , I have made the changes.

**Comment by razzeee** [comment]
I'm wondering if we should be using a `background-image` like the old design as the new one seems to be using an `img` tag.

New
image

Old
image


They also seem to handle the rating differently (maybe it's rounding differently?)

**Comment by razzeee** [comment]
Studio and Cast should also switch order

File: src/WSDecoder.elm [hunk 3]
```diff
@@ -770,6 +797,24 @@ type alias TvshowObj =
     , dateadded : String
     , rating : Float
     , thumbnail : String
+    , genre : List String
+    , mpaa : String
+    , plot : String
+    , season : Int
+    , studio : List String
+    , episode : Int
+    , watchepisode : Int
```
Comment (razzeee) [inline src/WSDecoder.elm:806]: Typo? Should be `watchedepisodes`?

**Comment by developingright** [comment]
Hi @razzeee,
I have fixed the typo in the TvShow Obj, and switched the order of cast and studio details. The rating was rendered on the front-end exactly how its value was unlike Chorus 2 which rounds the values.  Now, it's working the same. 

Also, about whether we should use `background-image` or `img`, I think it should be fine with `img` tag as the convention which is normally followed is that if the same image is used at multiple places its better to use background-image but in our case, we are fetching the fanart every time as it's different for every other object.

**Comment by razzeee** [comment]
We stll need to fix the image sizing then.

**Comment by developingright** [comment]
Hi @razzeee,
This issue can be considered a UI irregularity and as I suggested earlier, I was thinking of addressing these issues once the pages and functionalities have been implemented, in a separate PR. Please let me know if it’s fine with you. :slightly_smiling_face:

**Pull request merged**